lib/utils/sort-locales.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Given a list of locales, this utility will sort them by `fallback` order, i.e. the locales without fallbacks first,
+ * then the locales having them as fallbacks, and then recursively.
+ */
+export default function sortLocales (locales) {
+  const localeByFallback = {}
+
+  locales.forEach(locale => {
+    if (locale.fallbackCode === null) {
+      locale.fallbackCode = undefined
+    }
+
+    if (!localeByFallback[locale.fallbackCode]) {
+      localeByFallback[locale.fallbackCode] = []
+    }
+
+    localeByFallback[locale.fallbackCode].push(locale)
+  })
+
+  return sortByFallbackKey(undefined, localeByFallback)
+}
+
+function sortByFallbackKey (key, localeByFallback) {
+  if (!localeByFallback[key]) {
+    return []
+  } else {
```

Comment:
Maybe a nitpick, there's probably no need for the `else` branch since you'll be jumping out of the function if the condition above is true.

Reply:
Both work the same for me. So I removed the else, it has a more guarded style now, and yes, it feels closer aligned with the rest of the code base. :+1: